File: go/config/Wifi.go
package config

import (
	"github.com/lvcdy/huawei-lte-api-go/session"
)

// Wifi 对应 config/Wifi.py。
type Wifi struct {
	*session.ApiGroup
}

// NewWifi 创建 Wifi API 分组。
func NewWifi(s *session.Session) *Wifi {
	return &Wifi{ApiGroup: session.NewApiGroup(s)}
}

// Config 对应 wifi/config.xml。
func (w *Wifi) Config() (map[string]interface{}, error) {
	return w.S.Get("wifi/config.xml", nil, "config")
}

// Configure 对应 wifi/configure.xml。
func (w *Wifi) Configure() (map[string]interface{}, error) {
	return w.S.Get("wifi/configure.xml", nil, "config")
}

// CountryChannel 对应 wifi/countryChannel.xml。
func (w *Wifi) CountryChannel() (map[string]interface{}, error) {
	return w.S.Get("wifi/countryChannel.xml", nil, "config")
}

// ChannelAutoMatchHardware 对应 wifi/channelAutoMatchHardware.xml。
func (w *Wifi) ChannelAutoMatchHardware() (map[string]interface{}, error) {
	return w.S.Get("wifi/channelAutoMatchHardware.xml", nil, "config")
}

File: src/config/wifi.rs
//! Wifi config group (`config/Wifi.py`).

use serde_json::Value;

use crate::api::get_value;
use crate::connection::Connection;
use crate::errors::Result;

/// Wifi config group.
pub struct WifiConfig<'a> {
    conn: &'a Connection,
}

impl<'a> WifiConfig<'a> {
    /// Build a group bound to `conn`.
    pub fn new(conn: &'a Connection) -> Self {
        WifiConfig { conn }
    }

    /// `wifi/config.xml`. WiFi configuration.
    pub fn config(&self) -> Result<Value> {
        get_value(self.conn, "config/wifi/config.xml")
    }

    /// `wifi/configure.xml`. WiFi configuration.
    pub fn configure(&self) -> Result<Value> {
        get_value(self.conn, "config/wifi/configure.xml")
    }

    /// `wifi/countryChannel.xml`. WiFi country channel.
    pub fn country_channel(&self) -> Result<Value> {
        get_value(self.conn, "config/wifi/countryChannel.xml")
    }

    /// `wifi/channelAutoMatchHardware.xml`. Channel auto-match hardware status.
    pub fn channel_auto_match_hardware(&self) -> Result<Value> {
        get_value(self.conn, "config/wifi/channelAutoMatchHardware.xml")
    }
}
